src/components/ToggleCollapse.tsx

```diff
@@ -1,7 +1,8 @@
 import { CaretDownFilled, CaretUpFilled } from "@ant-design/icons";
-import { Switch } from "antd";
+import { Checkbox, Switch } from "antd";
```

Comment:
`ToggleCollapse` got some major updates!
- I was running into a bug where the height wouldn't update if the inner content changes size, which happens if you expand/collapse the individual channels. I got around this by making an `isAnimating` flag in state (and replacing a few related state variables in the process.)
  - While animating is happening, the height CSS attribute has a transition applied. Otherwise, there is no animation, which allows instant resizing if child content resizes.
- I added an option to replace the toggle with a checkbox.